### PR TITLE
Reorg: Move debug-helper to a plugin

### PR DESCRIPTION
**Pull request by kraftbj**
The `debug-helper` is in the `packages` dir right now, but it isn't really a composer package -- it is a plugin used for development purposes.

This would move it as a plugin.

#### Changes proposed in this Pull Request:
* moves debug to a plugin
* updates Docker to remove the special case for it (all plugin folders are added already).
* Adds composer.json file to prepare for the changes made in #18313.

#### Jetpack product discussion
- n/a
- 
#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Run `yarn docker:build-image && yarn docker:up`.
* Verify the debug plugin is a plugin on the install.

#### Proposed changelog entry for your changes:

*

### CODE REVIEW

File: projects/plugins/debug-helper/composer.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+	"name": "automattic/jetpack-debug-helper",
+	"description": "A plugin for mocking various states of the plugin to aid in Jetpack development. Not for production use.",
+	"type": "wordpress-plugin",
+	"license": "GPL-2.0-or-later",
+	"minimum-stability": "dev",
+	"require": {},
+	"extra": {
+		"mirror-repo": "Automattic/jetpack-debug-helper"
+	}
```
Comment (anomiex) [inline projects/plugins/debug-helper/composer.json:10]: The name in composer.json is almost always going to match the GitHub repo name, as is the case here. Specifying it is unnecessary, but doesn't hurt anything other than maybe confusing people into thinking this is required for something.

**Review by anomiex — COMMENTED** [review]
Some linter errors need cleaning up. Otherwise seems ok, haven't tested yet.

File: projects/plugins/debug-helper/composer.json (hunk 1 above)
Comment (kraftbj) [reply projects/plugins/debug-helper/composer.json:10]: I was wanting to err on the side of caution against overriding an existing, unrelated repo (and thus make `mirror-repo` an expected, if not required, field). If you think that's overkill, we can drop it.

Comment (anomiex) [reply projects/plugins/debug-helper/composer.json:10]: Let's see what happens with #18316. Or skip it here but then add it to #18316 once this is merged.

Comment (kraftbj) [reply projects/plugins/debug-helper/composer.json]: Removed from this PR.

**Review by anomiex — APPROVED** [review]

**Pull request merged**